src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDateArray = [];
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
```

Comment:
There is no validation to ensure that the `dateArray` contains the correct number of elements after splitting by the separator. If the date string is not in the expected format or if the separator is incorrect, this could result in an array with an incorrect number of elements, leading to potential errors in subsequent operations.